### PR TITLE
Add test to show issues from a script unloaded too late

### PR DESCRIPTION
**Pull request by Quetzacoalt91**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Change the order we call the different method when loading a new page on the new UI. We expect to call the `beforeDestroy` before updating the DOM, so we can safely remove stuff such as the event listeners.
| Type?             | improvement
| BC breaks?        | Nope
| Deprecations?     | Nope
| Fixed ticket?     | /
| Sponsor company   | @PrestaShopCorp
| How to test?      | There is no visible impact on the current state of the dev branch. But when implementing the next pages, we won't get unexpected exception for a missing element in the DOM while unloading a script.

**Note:** The first commit of this PR will fail. Only the test has been added to show how it will be fixed by the second commit.

### CODE REVIEW

**Review by ga-devfront — APPROVED** [review]

**Comment by Quetzacoalt91** [comment]
This PR is required for further changes on #986. Merging.

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**
